cfme/intelligence/reports/reports.py

```diff
@@ -312,6 +313,42 @@ def delete(self, cancel=False):
     def saved_reports(self):
         return self.collections.saved_reports
 
+    def create_schedule(self, name=None, description=None, active=None,
+            timer=None, emails=None, email_options=None):
+
+        view = navigate_to(self, "ScheduleReport")
+        view.fill({
+            "name": name,
+            "description": description,
+            "active": active,
+            "run": timer.get("run"),
+            "time_zone": timer.get("time_zone"),
+            "starting_date": timer.get("starting_date"),
+            "hour": timer.get("hour"),
+            "minute": timer.get("minute"),
+            "emails_send": bool(emails),
+            "emails": emails,
+            "send_if_empty": email_options.get("send_if_empty"),
+            "send_txt": email_options.get("send_txt"),
+            "send_csv": email_options.get("send_csv"),
+            "send_pdf": email_options.get("send_pdf")
+        })
+
+        view.add_button.click()
+        view.flash.assert_no_error()
+
+        schedule = self.appliance.collections.schedules.instantiate(
+            name=name or self.menu_name,
```

Comment:
Question:
can you please explain this?

Reply:
This instantiates the newly created schedule and returns it.